api/src/main/java/org/apache/brooklyn/api/mgmt/ManagementContext.java

```diff
@@ -66,9 +67,20 @@
      * In other words the value of {@link Application#getManagementContext()#getManagementPlaneId()} 
      * will generally be constant (in contrast to {@link #getManagementNodeId()}).
      * <p>
-     * This value should not be null unless the management context is a non-functional
-     * (non-deployment) instance. */
+     * Throws an {@link NullPointerException} if the value hasn't been initialized yet. The value is set:
```

Comment:
Wonder whether we should leave this to return `null` when not initialised. It's been useful to find instances of wrong (too early) usage but on the other hand changes the method behaviour.